From 533c229899364a84f653c5c3b55e888ace661b3a Mon Sep 17 00:00:00 2001
From: Mark Enriquez <enriquezmark36@gmail.com>
Date: Fri, 15 May 2020 17:26:40 +0800
Subject: [PATCH 3/4] sepolicy: Update policies; Remove mmcblk0 labels

This patch should update most rules and fixes some warnings due
to having some files in dos line endings.

mmcblk0p* labels are removed simply because not all SPRD devices
have the same partition layout and causes some mislablings that still
causes some AVC denials most notably to the fsck proccess.

Move this type of block labelings to the device's policy.

Change-Id: I845373ae5b65a4c0ae09b10158cb5afdfac1866c
Signed-off-by: Mark Enriquez <enriquezmark36@gmail.com>
---
 sepolicy/file_contexts    | 54 ++++++++++++++++++---------------------
 sepolicy/mediacodec.te    | 3 ++-
 1 files changed, 27 insertions(+), 30 deletions(-)

diff --git a/sepolicy/file_contexts b/sepolicy/file_contexts
index 671da21..153c639 100644
--- a/sepolicy/file_contexts
+++ b/sepolicy/file_contexts
@@ -28,30 +28,18 @@
 
 /sys/power/restart_cpc       u:object_r:sysfs_restart_cpc:s0
 
-/system/bin/at_distributor   u:object_r:rild_exec:s0
-/system/bin/modemd           u:object_r:rild_exec:s0
-/system/bin/nvitemd          u:object_r:nvitemd_exec:s0
-/system/bin/phoneserver      u:object_r:rild_exec:s0
-/system/bin/refnotify        u:object_r:refnotify_exec:s0
-/system/bin/ext_data.sh      u:object_r:rild_exec:s0
-/system/bin/ext_kill.sh      u:object_r:rild_exec:s0
-/system/bin/ext_symlink.sh   u:object_r:rild_exec:s0
+/system/vendor/bin/at_distributor   u:object_r:rild_exec:s0
+/system/vendor/bin/modemd           u:object_r:modemd_exec:s0
+/system/vendor/bin/nvitemd          u:object_r:nvitemd_exec:s0
+/system/vendor/bin/phoneserver      u:object_r:phoneserver_exec:s0
+/system/vendor/bin/refnotify        u:object_r:refnotify_exec:s0
+/system/vendor/bin/ext_data.sh      u:object_r:rild_exec:s0
+/system/vendor/bin/ext_kill.sh      u:object_r:rild_exec:s0
+/system/vendor/bin/ext_symlink.sh   u:object_r:rild_exec:s0
 
 /data/misc/radio(/.*)?       u:object_r:radio_data_file:s0
 
 # Block labeling
-/dev/block/mmcblk0p1         u:object_r:radio_block_device:s0
-/dev/block/mmcblk0p2         u:object_r:radio_block_device:s0
-/dev/block/mmcblk0p3         u:object_r:radio_block_device:s0
-/dev/block/mmcblk0p4         u:object_r:radio_block_device:s0
-
-/dev/block/mmcblk0p5         u:object_r:prod_block_device:s0
-/dev/block/mmcblk0p7         u:object_r:efs_block_device:s0
-/dev/block/mmcblk0p10        u:object_r:param_block_device:s0
-/dev/block/mmcblk0p17        u:object_r:cache_block_device:s0
-/dev/block/mmcblk0p18        u:object_r:system_block_device:s0
-/dev/block/mmcblk0p20        u:object_r:userdata_block_device:s0
-
 /dev/block/zram[0-9]+        u:object_r:swap_block_device:s0
 
 # EFS
@@ -60,7 +48,7 @@
 # WiFi
 /dev/rfkill                  u:object_r:rfkill_device:s0
 
-/system/bin/macloader        u:object_r:macloader_exec:s0
+/system/vendor/bin/macloader        u:object_r:macloader_exec:s0
 
 /data/.cid.info              u:object_r:wifi_data_file:s0
 /efs/wifi(/.*)?              u:object_r:wifi_data_file:s0
@@ -76,7 +64,7 @@
 
 /data/system/gps(/\.*.*)?    u:object_r:gps_data_file:s0
 
-/system/bin/gpsd             u:object_r:gpsd_exec:s0
+/system/vendor/bin/gpsd             u:object_r:gpsd_exec:s0
 
 # GFX
 /dev/mali[0-9]*              u:object_r:gpu_device:s0
@@ -84,20 +72,28 @@
 # Media
 /dev/pipe(/.*)?              u:object_r:media_device:s0
 
-/dev/sprd_dma_copy           u:object_r:audio_device:s0
-/dev/sprd_jpg                u:object_r:audio_device:s0
-/dev/sprd_rotation           u:object_r:audio_device:s0
-/dev/sprd_scale              u:object_r:audio_device:s0
-/dev/sprd_sensor             u:object_r:audio_device:s0
-/dev/sprd_vsp                u:object_r:audio_device:s0
+/dev/sprd_dma_copy           u:object_r:camera_device:s0
+/dev/sprd_jpg                u:object_r:camera_device:s0
+/dev/sprd_rotation           u:object_r:camera_device:s0
+/dev/sprd_scale              u:object_r:camera_device:s0
+/dev/sprd_sensor             u:object_r:camera_device:s0
+/dev/sprd_vsp                u:object_r:video_device:s0
 
 /data/local/media(/.*)?      u:object_r:media_data_file:s0
 
+
+/data/\.wifiver\.info          u:object_r:wifi_data_file:s0
 # ENGPC
-/system/bin/engpc            u:object_r:engpc_exec:s0
+/system/vendor/bin/engpc            u:object_r:engpc_exec:s0
 
 # Gadget Serial
 /dev/ttyGS[0-9]+             u:object_r:gadget_serial_device:s0
 
 # Sysfs
 /sys/devices/platform/sprd_fb.0/graphics/fb0/dynamic_pclk    u:object_r:sysfs_dynamic_pclk:s0
+/sys/devices/platform/scxx30-dmcfreq.0/devfreq/scxx30-dmcfreq.0/ondemand/set_freq  u:object_r:sysfs_dmcfreq_set_freq:s0
+/sys/power/wait_for_fb_sleep                                 u:object_r:sysfs_wait_for_fb:s0
+/sys/power/wait_for_fb_wake                                  u:object_r:sysfs_wait_for_fb:s0
+
+# Power hal has to have a SELinux context with labels
+/system/vendor/bin/hw/android\.hardware\.power@1\.0-service\.sc8830   u:object_r:hal_power_default_exec:s0
diff --git a/sepolicy/mediacodec.te b/sepolicy/mediacodec.te
index 22bc0b4..1c78de8 100644
--- a/sepolicy/mediacodec.te
+++ b/sepolicy/mediacodec.te
@@ -1,3 +1,4 @@
 set_prop(mediacodec, media_prop)
+get_prop(mediacodec, media_prop)
+
 allow mediacodec audio_device:chr_file { read write open ioctl };
-allow mediacodec system_file:file execmod;
\ No newline at end of file
-- 
2.28.0

